sdk/storage/azure-storage-blob/azure/storage/blob/_upload_helpers.py

```diff
@@ -93,17 +97,19 @@ def upload_block_blob(  # pylint: disable=too-many-locals, too-many-statements
         # Do single put if the size is smaller than or equal config.max_single_put_size
         if adjusted_count is not None and (adjusted_count <= blob_settings.max_single_put_size):
             try:
-                data = data.read(length)
+                data = data.read(length)  #type: ignore
                 if not isinstance(data, bytes):
                     raise TypeError('Blob data should be of type bytes.')
             except AttributeError:
                 pass
             if encryption_options.get('key'):
+                if not isinstance(data, bytes):
+                    raise TypeError('Blob data should be of type bytes.')
                 encryption_data, data = encrypt_blob(data, encryption_options['key'], encryption_options['version'])
                 headers['x-ms-meta-encryptiondata'] = encryption_data
 
             response = client.upload(
-                body=data,
+                body=data,  #type: ignore
```

Comment:
azure\storage\blob\_upload_helpers.py:112: error: Argument "body" to "upload" of "BlockBlobOperations" has incompatible type "Union[bytes, str, Iterable[str], IO[str]]"; expected "IO[Any]"  [arg-type]
azure\storage\blob\_upload_helpers.py:112: error: Argument "body" to "upload" of "BlockBlobOperations" has incompatible type "Union[bytes, str, Iterable[bytes], 

Again an issue with the typehint of `data`